src/cascadia/TerminalApp/DefaultProfileUtils.cpp

```diff
@@ -15,19 +15,20 @@ static constexpr std::wstring_view PACKAGED_PROFILE_ICON_EXTENSION{ L".png" };
 // - name: the name of the new profile.
 // Return Value:
 // - A Profile, ready to be filled in
-TerminalApp::Profile CreateDefaultProfile(const std::wstring_view name)
+winrt::TerminalApp::Profile CreateDefaultProfile(const std::wstring_view name)
 {
-    const auto profileGuid{ Microsoft::Console::Utils::CreateV5Uuid(TERMINAL_PROFILE_NAMESPACE_GUID,
+    const winrt::guid profileGuid{ Microsoft::Console::Utils::CreateV5Uuid(TERMINAL_PROFILE_NAMESPACE_GUID,

```

Comment:
this could still be `auto`, right? There should be an magic `GUID`->`winrt::guid` conversion already

Reply:
I wish. I think that the compiler gets confused with the `winrt::Windows::Foundation::IReference<winrt::guid>`. :(

```
1>D:\Terminal\src\cascadia\TerminalApp\DefaultProfileUtils.cpp(22,55): error C2440: 'initializing': cannot convert from 'initializer list' to 'winrt::TerminalApp::Profile'
1>D:\Terminal\src\cascadia\TerminalApp\DefaultProfileUtils.cpp(22,55): message : No constructor could take the source type, or constructor overload resolution was ambiguous
1>D:\Terminal\src\cascadia\TerminalApp\DefaultProfileUtils.cpp(24,32): error C2664: 'auto winrt::impl::consume_TerminalApp_IProfile<winrt::TerminalApp::IProfile>::Guid(const winrt::Windows::Foundation::IReference<winrt::guid> &) const': cannot convert argument 1 from 'const GUID' to 'const winrt::Windows::Foundation::IReference<winrt::guid> &'
1>D:\Terminal\src\cascadia\TerminalApp\DefaultProfileUtils.cpp(24,32): message : Reason: cannot convert from 'const GUID' to 'const winrt::Windows::Foundation::IReference<winrt::guid>'
1>D:\Terminal\src\cascadia\TerminalApp\DefaultProfileUtils.cpp(24,21): message : No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
```

Reply:
This actually seems to be the source of a lot of the comments you made. I've been getting around it by just using `winrt::guid` instead of `GUID`, but if you have a better idea in mind, go for it.

Reply:
oh that makes me unreasonably angry

Reply:
something something type covariance